### PR TITLE
feat: Replace flask with quart

### LINKED ISSUE #42: Runner crash on first request on Linux (and maybe others)
### Describe the bug

The runner crashes on first request (usually program message)

### To Reproduce

1. Open VSCode with Safe-DS extension
2. Click on Run button for any Safe-DS source file

### Expected behavior

The runner executes the program

### Screenshots (optional)

_No response_

### Additional Context (optional)

I would propose to replace the (legacy) flask stack with a quart (successor in a way of flask) stack that does not require monkey patching python internals. This has the added benefit to allow async python functions and in turn being possibly more efficient.

Furthermore, a migration to the socket.io protocol is still possible although not as easy as changing the dependency in flask.

### PR DESCRIPTION
**Pull request by WinPlay02**
Closes #42 

### Summary of Changes

- replaces flask with quart
- rewrite some tests
- added test against crashing
- solution should be more scalable as it replaces wsgi with asgi

### CODE REVIEW

**Review by SmiteDeluxe — DISMISSED** [review]
Tested locally with my eda branch and it is working and fixing all my issues! Have not looked at the code for specifics though, @lars-reimann is probably on that!

File: src/safeds_runner/server/main.py [hunk 1]
```diff
@@ -14,15 +14,6 @@ def start_server(port: int) -> None:
...
-    from gevent.monkey import patch_all
 
-    # Patch WebSockets to work in parallel
-    patch_all()
-
-    from safeds_runner.server.server import SafeDsServer
-
-    safeds_server = SafeDsServer(app_pipeline_manager)  # pragma: no cover
+    safeds_server = SafeDsServer()  # pragma: no cover
     safeds_server.listen(port)  # pragma: no cover
```
Comment (lars-reimann) [inline src/safeds_runner/server/main.py:19]: Is the `# pragma: no cover` really needed here? It seems like if line 16 is run, so are the subsequent lines.

Comment (WinPlay02) [reply src/safeds_runner/server/main.py:19]: I agree, line 18 should be covered. However, line 19 does never return, so it should stay that way. 
- If successful, the eventloop runs forever
- If unsuccessful, the stack unwinds
- If a shutdown was requested, the runner exits via a syscall

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**